packages/tools/src/lib/build-cache/common.ts

```diff
@@ -2,36 +2,63 @@ import fs from 'node:fs';
 import path from 'node:path';
 import { nativeFingerprint } from '../fingerprint/index.js';
 import { getCacheRootPath } from '../project.js';
-import type { spinner } from '../prompts.js';
 
 export const BUILD_CACHE_DIR = 'remote-build';
 
 export type SupportedRemoteCacheProviders = 'github-actions';
 
 export type RemoteArtifact = {
   name: string;
-  downloadUrl: string;
+  url: string;
 };
 
 export type LocalArtifact = {
   name: string;
-  path: string;
 };
 
+/**
+ * Interface for implementing remote build cache providers.
+ * Remote cache providers allow storing and retrieving native build artifacts (e.g. APK, IPA)
+ * from remote storage like S3, GitHub Artifacts etc.
+ */
 export interface RemoteBuildCache {
+  /** Unique identifier for this cache provider, will be displayed in logs */
   name: string;
-  query({
+
+  /**
+   * List available artifacts matching the given name pattern
+   * @param artifactName - Passed after fingerprinting the build, e.g. `rnef-android-debug-1234567890` for android in debug variant
+   * @param limit - Optional maximum number of artifacts to return
+   * @returns Array of matching remote artifacts, or empty array if none found
+   */
+  list({
     artifactName,
+    limit,
   }: {
-    artifactName: string;
-  }): Promise<RemoteArtifact | null>;
-  download({
-    artifact,
-    loader,
-  }: {
-    artifact: RemoteArtifact;
-    loader: ReturnType<typeof spinner>;
-  }): Promise<LocalArtifact>;
+    artifactName: string | undefined;
+    limit?: number;
+  }): Promise<RemoteArtifact[]>;
+
+  /**
+   * Download a remote artifact to local storage
+   * @param artifactName - Name of the artifact to download, e.g. `rnef-android-debug-1234567890` for android in debug variant
+   * @returns Response object from fetch, which will be used to download the artifact
+   */
+  download({ artifactName }: { artifactName: string }): Promise<Response>;
+
+  /**
+   * Delete a remote artifact
+   * @param artifact - Remote artifact to delete, as returned by `list` method
+   * Throws if artifact is not found or deletion fails
+   */
+  delete({ artifactName }: { artifactName: string }): Promise<RemoteArtifact[]>;
```

Comment:
let's either adjust the comment above or remove the return type here. also we could use: 

`@throws {Error} Throws if artifact is not found or deletion fails` syntax which points to the Error type that will be thrown